src/webgl/p5.Camera.js

```diff
@@ -0,0 +1,953 @@
+ * orientation.
+ * @method setPosition
+ * @param {Number} x x position of a point in world space
+ * @param {Number} y y position of a point in world space
+ * @param {Number} z z position of a point in world space
+ * @example
+ * <div>
+ * <code>
+ * // press '1' '2' or '3' keys to set camera position
+ *
+ * var cam;
+ *
+ * function setup() {
+ *   createCanvas(100, 100, WEBGL);
+ *   normalMaterial();
+ *   cam = createCamera();
+ * }
+ *
+ * function draw() {
+ *   background(200);
+ *
+ *   // '1' key
+ *   if (keyIsDown(49)) {
+ *     cam.setPosition(30, 0, 80);
+ *   }
+ *   // '2' key
+ *   if (keyIsDown(50)) {
+ *     cam.setPosition(0, 0, 80);
+ *   }
+ *   // '3' key
+ *   if (keyIsDown(51)) {
+ *     cam.setPosition(-30, 0, 80);
+ *   }
+ *
+ *   box(20);
+ * }
+ * </code>
+ * </div>
+ *
+ * @alt
+ * camera position changes as the user presses keys, altering view of a 3D box
+ */
+p5.Camera.prototype.setPosition = function(x, y, z) {
+  var diffX = x - this.eyeX;
+  var diffY = y - this.eyeY;
+  var diffZ = z - this.eyeZ;
+
+  this.camera(
+    x,
+    y,
+    z,
+    this.centerX + diffX,
+    this.centerY + diffY,
+    this.centerZ + diffZ,
+    0,
+    1,
+    0
+  );
+};
+
+////////////////////////////////////////////////////////////////////////////////
+// Camera Helper Methods
+////////////////////////////////////////////////////////////////////////////////
+
+// @TODO: combine this function with _setDefaultCamera to compute these values
+// as-needed
+p5.Camera.prototype._computeCameraDefaultSettings = function() {
+  this.defaultCameraFOV = 60 / 180 * Math.PI;
+  this.defaultAspectRatio = this._renderer.width / this._renderer.height;
+  this.defaultEyeX = 0;
+  this.defaultEyeY = 0;
+  this.defaultEyeZ =
+    this._renderer.height / 2.0 / Math.tan(this.defaultCameraFOV / 2.0);
+  this.defaultCenterX = 0;
+  this.defaultCenterY = 0;
+  this.defaultCenterZ = 0;
+  this.defaultCameraNear = this.defaultEyeZ * 0.1;
+  this.defaultCameraFar = this.defaultEyeZ * 10;
+};
+
+//detect if user didn't set the camera
+//then call this function below
+p5.Camera.prototype._setDefaultCamera = function() {
+  this.cameraFOV = this.defaultCameraFOV;
+  this.aspectRatio = this.defaultAspectRatio;
+  this.eyeX = this.defaultEyeX;
+  this.eyeY = this.defaultEyeY;
+  this.eyeZ = this.defaultEyeZ;
+  this.centerX = this.defaultCenterX;
+  this.centerY = this.defaultCenterY;
+  this.centerZ = this.defaultCenterZ;
+  this.upX = 0;
+  this.upY = 1;
+  this.upZ = 0;
+  this.cameraNear = this.defaultCameraNear;
+  this.cameraFar = this.defaultCameraFar;
+
+  this.perspective();
+  this.camera();
+
+  this.cameraType = 'default';
+};
+
+p5.Camera.prototype._resize = function() {
+  // If we're using the default camera, update the aspect ratio
+  if (this.cameraType === 'default') {
+    this._computeCameraDefaultSettings();
+    this._setDefaultCamera();
+  } else {
+    this.defaultAspectRatio = this._renderer.width / this._renderer.height;
```

Comment:
will this always overwrite the user's camera with a default one if the window is resized? what's the right thing to do here?

Reply:
Yes, that is a mistake.  That else statement should call:
```
this.perspective(
      this.cameraFOV,
      this._renderer.width / this._renderer.height
    );
``` 
and really doesn't need to call `this.camera()` as the cameraMatrix isn't affected by aspect ratio.  I think default behavior for resize could be one of two things:
1. adjust only aspect ratio on a resize event (such that nothing is squished/stretched),
2. adjust aspect ratio & eyeZ so that objects retain the same relative size on canvas.  This is what is happening for the default cameras.

I don't see a great reason to have eyeZ change on resize events (for either the custom or default cameras), and think having consistent behavior for custom / default cameras would be simpler. Thoughts?